contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -122,35 +132,102 @@ def validate_presence_of_filter_macro(
+        # convert these to ManualTest
+        tmp: list[UnitTest | IntegrationTest | ManualTest] = []
+        if self.tags.manual_test is not None:
+            for test in self.tests:
+                if not isinstance(test, UnitTest):
+                    raise ValueError(
+                        "At this point of intialization, tests should only be UnitTest objects, "
+                        f"but encountered a {type(test)}."
+                    )
+                # Create the manual test and skip it upon creation (cannot test via contentctl)
+                manual_test = ManualTest(
+                    name=test.name,
+                    attack_data=test.attack_data
+                )
+                tmp.append(manual_test)
+            self.tests = tmp
 
-        # iterate over the unit tests and create a TestGroup (and as a result, an IntegrationTest) for each
-        test_groups: list[TestGroup] = []
-        tests: list[UnitTest | IntegrationTest] = info.data.get("tests")                            # type: ignore
-        unit_test: UnitTest
-        for unit_test in tests:                                                                     # type: ignore
-            test_group = TestGroup.derive_from_unit_test(unit_test, info.data.get("name"))          # type: ignore
-            test_groups.append(test_group)
+        # iterate over the tests and create a TestGroup (and as a result, an IntegrationTest) for
+        # each unit test
+        self.test_groups = []
+        for test in self.tests:
+            # We only derive TestGroups from UnitTests (ManualTest is ignored and IntegrationTests
+            # have not been created yet)
+            if isinstance(test, UnitTest):
+                test_group = TestGroup.derive_from_unit_test(test, self.name)
+                self.test_groups.append(test_group)
 
         # now add each integration test to the list of tests
-        for test_group in test_groups:
-            tests.append(test_group.integration_test)
-        return test_groups
+        for test_group in self.test_groups:
+            self.tests.append(test_group.integration_test)
+
+        # Skip all manual tests
+        self.skip_manual_tests()
+
+        # NOTE: we ignore the type error around self.status because we are using Pydantic's
+        # use_enum_values configuration
+        # https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.populate_by_name
+
+        # Skip tests for non-production detections
+        if self.status != DetectionStatus.production.value:                                         # type: ignore
+            self.skip_all_tests(f"TEST SKIPPED: Detection is non-production ({self.status})")
+
+        # TODO (cmcginley): should we just mark all Correlation type detections as manual_test?
+        # Skip tests for detecton types like Correlation which are not supported via contentctl
+        if self.type in UNTESTED_ANALYTICS_TYPES:
```

Comment:
Should we maybe handle CorrelationSearch detections differently? Instead of adding special handling for them in contentctl, why not just mark them all as `manual_test` in ESCU?

Reply:
We WILL begin testing CorrelationSearches soon (you and I can discuss this in more detail offline). If that's the case, I think it will be a good idea to remove it from UNTESTED_ANALYTICS_TYPES and all those Correlation Searches will begin failing validation since they don't have tests (rather than needing to update the manual_test field in EVERY detection which now _should_ have a test).

Reply:
Ok cool, that's good context. We probably can't disable this check until we have a PR ready which gives them test cases. If we disable this check before that, we won't be able to test any content since validation will fail and testing will never be reached

Reply:
Let's leave this check in place now, without adding manual_test, as discussed based on future work to test this content.